src/Components/Blazor/Build/src/Core/RuntimeDependenciesResolver.cs

```diff
@@ -108,58 +110,65 @@ void ResolveAssembliesCore()
                     }
                 }
 
-                IEnumerable<AssemblyNameReference> GetAssemblyReferences(AssemblyEntry current) =>
-                    current.Definition.Modules.SelectMany(m => m.AssemblyReferences);
-
-                AssemblyEntry Resolve(AssemblyNameReference current)
+                AssemblyEntry Resolve(string assemblyName)
                 {
-                    if (Assembly.Definition.Name.Name == current.Name)
+                    if (Assembly.Name == assemblyName)
                     {
                         return Assembly;
                     }
 
-                    var referencedAssemblyCandidate = FindCandidate(current, Dependencies);
-                    var bclAssemblyCandidate = FindCandidate(current, Bcl);
-
                     // Resolution logic. For right now, we will prefer the mono BCL version of a given
                     // assembly if there is a candidate assembly and an equivalent mono assembly.
-                    if (bclAssemblyCandidate != null)
-                    {
-                        return bclAssemblyCandidate;
-                    }
-
-                    return referencedAssemblyCandidate;
+                    return Bcl.FirstOrDefault(c => c.Name == assemblyName) ??
+                        Dependencies.FirstOrDefault(c => c.Name == assemblyName);
```

Comment:
Do you think the O(N^2)-ness of this could become an issue if there's a really large number of dependencies? Is it work making `Bcl` and `Dependencies` into a dictionary mapping name to `AssemblyEntry`?

I'm not insisting this should be done; just asking what you think.